src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
 }
 
-export class Triangle {
+export class Rectangle implements Figure {
+  public shape: Shape = 'rectangle';
 
+  constructor(
+    public color: Color,
+    public height: number,
+    public width: number,
+  ) {
+    if (
+      this.height < 0 || this.width < 0
+    ) {
+      throw new Error('Insert correct data');
```

Comment:
(checklist) [CODE STYLE] - If you throw an error try to provide its detailed description in error message.